lib/navigation.php

```diff
@@ -517,3 +517,57 @@ function gutenberg_rename_navigation_menus_admin_menu_entry() {
 }
 
 add_action( 'admin_menu', 'gutenberg_rename_navigation_menus_admin_menu_entry' );
+
+/**
+ * Registers the navigation areas supported by the current theme. The expected
+ * shape of the argument is:
+ * array(
+ *     'primary'   => 'Primary',
+ *     'secondary' => 'Secondary',
+ *     'tertiary'  => 'Tertiary',
+ * )
+ *
+ * @param array $new_areas Supported navigation areas.
+ */
+function gutenberg_register_navigation_areas( $new_areas ) {
+	global $gutenberg_navigation_areas;
+	$gutenberg_navigation_areas = $new_areas;
+}
+
+// Register the default navigation areas.
+gutenberg_register_navigation_areas(
+	array(
+		'primary'   => 'Primary',
+		'secondary' => 'Secondary',
+		'tertiary'  => 'Tertiary',
+	)
+);
```

Comment:
Something I just noticed. If a theme dev is able to change the keys here by calling `register_navigation_areas` it'll degrade the capability of theme migration. This is a problem we have with the existing menus system in WordPress.

An example - one theme calls a menu location 'Header', the other 'Primary', and when switching no migration happens because the names don't match.

The way around this, I think, is to keep an internal representation of the area `primary`, `secondary` (or maybe even just an index `0`, `1`), but allow users to set a name that's used in the UI ('Header', 'Footer').

I quite like the idea of:
```php
	array(
		'primary'   => 'Header',
		'secondary' => 'Sidebar',
		'tertiary'  => 'Footer',
	)
```

But then what if I have four or five or ten navigation areas? What comes after tertiary?

It's a difficult problem, but I think we should try to solve it for 5.9 by improving this function. Lets come up with some ideas for this.

Reply:
I think we should provide some strong defaults but allow users to change extend how they like.

Most devs will stick with the convention that comes by default so that will solve 80% use case of Theme switch.

If a Theme dev choose to alter the area name then they will need to handle the situation where navigations don't migrate across on Theme switch. This provides a big incentive to stick with the registered defaults.

I didn't realise that `register_navigation_areas` would overwrite the existing items. If that's the case then I think we should make that function private and instead provide a filter. This approach is more "WordPress" and also tends to encourage extension rather than wholesale overwriting.


> What comes after tertiary?

`quaternary, quinary, senary, septenary, octonary, nonary, denary.`

Reply:
I've added a comment over here - https://github.com/WordPress/gutenberg/issues/36373#issuecomment-968511319